src/job/jobValidator.ts

```diff
@@ -0,0 +1,85 @@
+import { AuthorizationError, InvalidJobError } from "../errors/errors";
+import validator from "validator";
+import { IJob } from "../entities/job";
+import { IFileSystemServices } from "../services/fileServices";
+import { RepoEntitlementsRepository } from "../repositories/repoEntitlementsRepository";
+
+export interface IJobValidator {
+    throwIfJobInvalid(job: IJob): Promise<void>;
+    throwIfBranchNotConfigured(job: IJob): Promise<void>;
+    throwIfUserNotEntitled(job: IJob): Promise<void>;
+    throwIfItIsNotPublishable(job: IJob): void;
+}
+
+export class JobValidator implements IJobValidator {
+    _fileSystemService: IFileSystemServices;
+    _repoEntitlementRepository: RepoEntitlementsRepository;
+    constructor(fileSystemService: IFileSystemServices, repoEntitlementRepository: RepoEntitlementsRepository) {
+        this._fileSystemService = fileSystemService;
+        this._repoEntitlementRepository = repoEntitlementRepository;
+    }
+
+    async throwIfUserNotEntitled(job: IJob): Promise<void> {
```

Comment:
I know we have this in the existing code, but how would this ever happen? 

Folks deploy via /deploy, which is populated based on a user's entitlements. How could they try to deploy a repo that isn't listed in the /deploy modal?

Reply:
Yeah the idea is to make the code agnostic of the callers. if we dont check it here , then our code will be with an assumption that validations are done by caller. 
this assumption might impact us when we want to introduce an API to trigger the build where the payload will be provided by the caller directly( in this case it will be the writers). 

Said that, I am open for discussions on it

Reply:
Reasonable rationale. I'm not enthusiastic about the idea of providing an API through which writers deploy -- in fact, I'm actively against it, but I suppose there's no harm in double-checking.